centreon/www/class/centreonDowntime.class.php

```diff
@@ -265,211 +265,319 @@ public function getInfos($id)
+    public function getRelations(int $downtimeId): array
     {
-        $list = array(
-            "host" => array(),
-            "hostgrp" => array(),
-            "svc" => array(),
-            "svcgrp" => array()
-        );
-        foreach (array_keys($list) as $type) {
-            switch ($type) {
-                case 'host':
-                    $query = "SELECT host_host_id as obj_id FROM downtime_host_relation WHERE dt_id = :id";
+        $relations = [
+            'hosts' => [],
+            'hostgroups' => [],
+            'services' => [],
+            'servicegroups' => [],
+        ];
+
+        foreach (array_keys($relations) as $resourceType) {
+            switch ($resourceType) {
+                case 'hosts':
+                    $query = 'SELECT
+                        dhr.host_host_id AS resource_id,
+                        h.host_activate AS activated
+                    FROM downtime_host_relation dhr
+                    INNER JOIN host h
+                        ON dhr.host_host_id = h.host_id
+                    WHERE
+                        dt_id = :downtimeId';
                     break;
-                case 'hostgrp':
-                    $query = "SELECT hg_hg_id as obj_id FROM downtime_hostgroup_relation WHERE dt_id = :id";
+                case 'hostgroups':
+                    $query = 'SELECT
+                        dhgr.hg_hg_id AS resource_id,
+                        hg.hg_activate AS activated
+                    FROM downtime_hostgroup_relation dhgr
+                    INNER JOIN hostgroup hg
+                        ON dhgr.hg_hg_id = hg.hg_id
+                    WHERE
+                        dt_id = :downtimeId';
                     break;
-                case 'svc':
-                    $query = "SELECT CONCAT(host_host_id, CONCAT('-', service_service_id)) as obj_id " .
-                        "FROM downtime_service_relation WHERE dt_id = :id";
+                case 'services':
+                    $query = 'SELECT CONCAT(dsr.host_host_id, CONCAT(\'-\', dsr.service_service_id)) AS resource_id,
+                        s.service_activate AS activated
+                    FROM downtime_service_relation dsr
+                    INNER JOIN service s
+                        ON dsr.service_service_id = s.service_id
+                    WHERE
+                        dt_id = :downtimeId';
                     break;
-                case 'svcgrp':
-                    $query = "SELECT sg_sg_id as obj_id FROM downtime_servicegroup_relation WHERE dt_id = :id";
+                case 'servicegroups':
+                    $query = 'SELECT
+                        dsgr.sg_sg_id AS resource_id,
+                        sg.sg_activate AS activated
+                    FROM downtime_servicegroup_relation dsgr
+                    INNER JOIN servicegroup sg
+                        ON dsgr.sg_sg_id = sg.sg_id
+                    WHERE
+                        dt_id = :downtimeId';
```

Comment:
same as previous x3

Reply:
Okay I'll apply this thanks 💯

Reply:
Suggestion applied thanks for the review 👍